src/ui/modals/smithing-modal-window.js

```diff
@@ -62,48 +61,73 @@ export class SmithingModalWindow extends ModalWindow {
 
         switch (bar) {
             case "Bronze Bar":
-                    recipes = ["BronzeDagger", "BronzeSword", null, "BronzeScimitar", "BronzeAxe", "Bronze2hSword"];
+                recipes = [
+                    "BronzeDagger",
+                    "BronzeAxe",
+                    null,
+                    null,
+                    null,
+                    "BronzeSword",
+                    "BronzeScimitar",
+                    "Bronze2hSword",
+                ];
                 break;
             default:
                 return;
         }
 
-        let rowIndex = 0;
-
-        let positionX = this.x + this.leftOffset + this.horizontalPadding;
-        let positionY = this.y + this.topOffset + this.verticalPadding;
+        const offsetX = this.x + this.leftOffset + this.horizontalPadding;
+        const offsetY = this.y + this.topOffset + this.verticalPadding;
+        const padX = this.iconWidth + this.horizontalPadding;
+        const padY = this.iconHeight + this.verticalPadding + 24;
 
-        for (let recipe of recipes) {
+        for (let [index, recipe] of recipes.entries()) {
             if (!recipe) {
-                positionX += this.iconWidth + this.horizontalPadding;
-                rowIndex += 1;
                 continue;
             }
 
-            if (rowIndex >= this.maxIconsPerRow) {
-                positionX = this.x + this.leftOffset + this.horizontalPadding;
-                positionY += this.iconHeight + this.verticalPadding + 24;
-                rowIndex = 0;
-            }
+            const positionX = offsetX + padX * (index % this.maxIconsPerRow);
+            const positionY = offsetY + padY * parseInt(index / this.maxIconsPerRow);
 
             const item = await getItemClass(recipe, this.scene.window);
+            console.log(item.name);
```

Comment:
Is this leftover?

Reply:
Wait on me for what?

Reply:
I think we've done plenty on this PR already so anything else can probably be done later if needed

Reply:
I meant I was going to wait to see if you had any more changes in mind. I'll remove the console.log after I get off work today.